pkg/operators/machine-api-operator.go

```diff
@@ -164,3 +166,72 @@ var _ = Describe("[Feature:Operators] Machine API cluster operator status should
 		Expect(framework.IsStatusAvailable(client, "machine-api")).To(BeTrue())
 	})
 })
+
+var _ = Describe("[Feature:Operators] When cluster-wide proxy is configured, Machine API cluster operator should ", func() {
+	defer GinkgoRecover()
+	var machineSet *mapiv1beta1.MachineSet
+	var machineSetParams framework.MachineSetParams
+
+	It("deploy proxy", func() {
+		client, err := framework.LoadClient()
+		deployment, err := framework.DeployClusterProxy(client)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(framework.IsDeploymentAvailable(client, deployment.Name, deployment.Namespace)).To(BeTrue())
+	})
+
+	It("configure proxy", func() {
+		client, err := framework.LoadClient()
+		pods, err := framework.GetPods(client, map[string]string{"app": "squid"})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pods.Items) > 0).To(BeTrue())
+		proxy := v1.Proxy{}
+		proxy.Spec.HTTPProxy = "http://" + pods.Items[0].Status.HostIP + ":3128"
+		proxy.Spec.HTTPSProxy = "http://" + pods.Items[0].Status.HostIP + ":3128"
+		framework.SetClusterProxy(client, proxy)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	It("reflect proxy in machine-api-controller", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(framework.IsProxyInjectedForDeployment(client, maoManagedDeployment, framework.MachineAPINamespace, true)).To(BeTrue())
+	})
+
+	It("create a machineset", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		machineSetParams = framework.BuildMachineSetParams(client, 3)
+		machineSet, err = framework.CreateMachineSet(client, machineSetParams)
+		Expect(err).ToNot(HaveOccurred())
+		framework.WaitForMachineSet(client, machineSet.GetName())
+
+	})
+
+	It("destroy a machineset", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		err = client.Delete(context.Background(), machineSet)
+		Expect(err).ToNot(HaveOccurred())
+		framework.WaitForMachineSetDelete(client, machineSet)
+	})
+
+	It("disable cluster-wide proxy", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		proxy := v1.Proxy{}
+		framework.SetClusterProxy(client, proxy)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("reflect removal of proxy in machine-api-controller", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(framework.IsProxyInjectedForDeployment(client, maoManagedDeployment, framework.MachineAPINamespace, false)).To(BeTrue())
+	})
+
+	It("tear down cluster proxy", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		framework.DestroyClusterProxy(client)
+	})
```

Comment:
I think there's some confusion here over how the test framework ginkgo works. Each `It` constitutes a different test case, these look like steps in a single test case?

If so, make each of these `It`s a `By` and wrap them all in an `It`

eg

```
It("should be able to create a machineSet when a proxy is configured", func() {
  By("deploy proxy", func() {
    ... 
  })

  By("configure proxy", func() {
    ... 
  })
})

Reply:
makes total sense. I restructured in to a three test cases which test specific aspects.